### PR TITLE
[FLINK-26076] Fix ArchUnit violations in Source(Sink)MetricsITCase

### PR DESCRIPTION
**Pull request by dawidwys**
## What is the purpose of the change

Fix ArchUnit violations regarding InMemoryReporter

## Brief change log

* parameterize search methods with JobID instead of recreating the reporter (and cluster) for every test

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by dawidwys** [comment]
Do you mind taking a look @zentol @JingGe ?

**Review by JingGe — APPROVED** [review]
Thanks @dawidwys for driving this PR. LGTM.

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java [hunk 1]
```diff
@@ -170,6 +176,10 @@ private String getOperatorName(MetricGroup g) {
         return g.getScopeComponents()[g.getScopeComponents().length - 2];
     }
 
+    private String getJobId(MetricGroup g) {
+        return g.getScopeComponents()[0];
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java:180]: use `#getAllVariables().get(ScopeFormat.SCOPE_JOB_ID)`

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java [hunk 2]
```diff
@@ -240,6 +252,15 @@ public void addToConfiguration(Configuration configuration) {
         configuration.setString(
                 ConfigConstants.METRICS_REPORTER_PREFIX + "mini_cluster_resource_reporter." + ID,
                 id.toString());
+        configuration.set(SCOPE_NAMING_JM_JOB, "<job_id>.<host>.jobmanager.<job_name>");
```
Comment (zentol) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java:255]: if you rely on the variables map (see above) then this change is unnecessary

Comment (dawidwys) [reply flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InMemoryReporter.java:255]: Great! That's a much better approach!

**Review by zentol — COMMENTED** [review]
Generally fine but I think we can simplify it.

**Review by zentol — APPROVED** [review]

**Comment by dawidwys** [comment]
Passed on private azure: https://dev.azure.com/wysakowiczdawid/Flink/_build/results?buildId=1346&view=results

**Pull request closed without merging**